src/utilities/id_assignment.cpp

```diff
@@ -24,140 +14,158 @@ namespace adm {
     }
   }
 
-  template <typename It>
-  void reassignAudioProgrammeIds(It begin, It end) {
-    auto audioProgrammeIdValue = AudioProgrammeIdValue(0x1001u);
-    undefineIds(begin, end);
-    for (auto it = begin; it != end; ++it) {
-      auto audioProgramme = *it;
-      auto audioProgrammeId = audioProgramme->template get<AudioProgrammeId>();
+  void reassignIds(std::shared_ptr<Document> document) {
+    IdReassigner{document};
+  }
+
+  IdReassigner::IdReassigner(std::shared_ptr<Document> document)
+      : document{document} {
+    reassignAllIds();
+  }
+
+  void IdReassigner::reassignAllIds() {
+    reassignAudioProgrammeIds();
+    reassignAudioContentIds();
+    reassignAudioObjectIds();
+    reassignAudioPackFormatIds();
+    /**
+     * Initialize all audioTrackFormatIds and audioChanneldFormatIds to zero.
+     * The reason behind this is, that the reassignAudioStreamFormatIds
+     * algorithm only gives new IDs to audioTrackFormats and
+     * audioChannelFormats which are referenced by an audioStreamFormat. 
+     * Additionally, audioStreamFormats are only given a valid ID if they 
+     * reference an audioChannelFormat. This should be the right way to do it 
+     * for 2076-0/1 structures. Unreferenced elements will get an ID with 
+     * value 0 and are thereby marked as elements which should be ignored.
+     * For 2076-2 structures, reassignAudioTrackUidIds will apply a unique ID
+     * to audioChannelFormats referenced directly from audioTrackUids.
+     */
+    undefineIds(document->getElements<AudioTrackFormat>().begin(),
+                document->getElements<AudioTrackFormat>().end());
+    undefineIds(document->getElements<AudioChannelFormat>().begin(),
+                document->getElements<AudioChannelFormat>().end());
+    reassignAudioStreamFormatIds();
+    reassignAudioTrackUidIds();
+  }
+
+  void IdReassigner::reassignAudioProgrammeIds() {
+    auto audioProgrammes = document->getElements<AudioProgramme>();
+    undefineIds(audioProgrammes.begin(), audioProgrammes.end());
```

Comment:
Yes, because you can otherwise run into "ID in use" errors when an ID your trying to apply is already taken by another element which you haven't yet reassigned.